src/ert/parsing/lark_parser_error_info.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import InitVar, dataclass
+from typing import List, Optional, Union
+
+from .lark_parser_file_context_token import FileContextToken
+from .lark_parser_types import MaybeWithContext
+
+
+@dataclass
+# pylint: disable=too-many-instance-attributes
+class ErrorInfo:
+    message: str
+    filename: Optional[str]
+    start_pos: Optional[int] = None
+    line: Optional[int] = None
+    column: Optional[int] = None
+    end_line: Optional[int] = None
+    end_column: Optional[int] = None
+    end_pos: Optional[int] = None
+    originates_from: InitVar[Union[MaybeWithContext, FileContextToken]] = None
+    originates_from_these: InitVar[
+        List[Union[MaybeWithContext, FileContextToken]]
+    ] = None
+    originates_from_keyword: InitVar[Union[MaybeWithContext, FileContextToken]] = None
+
+    def __post_init__(
+        self,
+        originates_from: Optional[Union[MaybeWithContext, FileContextToken]],
```

Comment:
I think it would be easiert to instead of having three optional values that we branch on, that we use builder pattern to attack the token. So the call_site would do
```python
ErrorInfo(
 message=msg,
 filename=filename,
 ...
).set_context_list(context_list)
```

instead of
```python
ErrorInfo(
 message=msg,
 filename=filename,
 ...,
 originiates_from_these=context_list
)

Reply:
Done